crates/sui-types/src/effects/effects_v2.rs

```diff
@@ -411,7 +412,7 @@ impl TransactionEffectsV2 {
         executed_epoch: EpochId,
         gas_used: GasCostSummary,
         shared_objects: Vec<SharedInput>,
-        loaded_per_epoch_config_objects: BTreeSet<ObjectID>,
+        loaded_per_epoch_config_objects: BTreeMap<ObjectID, Option<SequenceNumber>>,
```

Comment:
I really really wish `Rust` had `Either` in the stdlib. 
Just as a small note, I feel like this should be `Either<BTreeSet<ObjectID>, BTreeMap<ObjectID, SequenceNumber>>`. I paused at first because I was confused under what situations we wouldn't have the number, but if I followed correctly, it is basically an `Either<(), SequenceNumber>` gated on the `ProtocolConfig`

At least maybe a comment or debug assert that they are all none or all some

Reply:
Yep! Followed exactly. This is really an `Either<BTreeSet<ObjectID>, BTreeMap<ObjectID, SequenceNumber>>`. I think comment + a debug assert would make sense here -- will add that in :)